### DISCUSSION TITLE
Help needed

### DISCUSSION THREAD

**Discussion by mikeh2010** [teslamate-org/teslamate#2755] [Q&A] [answered]
We just upgraded to the latest release and got the ENCRYPTION_KEY message. Restarted with 'docker-compose logs teslamate' to get the key and added it to the docker-compose.yml

I came back into the room to a worried face! apparently it wouldn't restart and so a --remove-orphans flag was run which toasted it.

Any thoughts on the best way to fix it? Found xterm contents pasted below...

```
root@docker-s-1vcpu-1gb-lon1-01:~# sudo docker-compose down
WARNING: Found orphan containers (teslamate_mqtt_api) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Removing network root_default
root@docker-s-1vcpu-1gb-lon1-01:~# sudo docker-compose down --remove-orphans
Removing orphan container "teslamate_mqtt_api"
Removing network root_default
WARNING: Network root_default not found.
root@docker-s-1vcpu-1gb-lon1-01:~# sudo docker-compose logs teslamate
Attaching to
root@docker-s-1vcpu-1gb-lon1-01:~# sudo docker-compose down
Removing network root_default
WARNING: Network root_default not found.
root@docker-s-1vcpu-1gb-lon1-01:~# docker-compose logs teslamate
Attaching to
root@docker-s-1vcpu-1gb-lon1-01:~# sudo docker-compose down
Removing network root_default
WARNING: Network root_default not found

```

**Comment by mikeh2010** [comment]
Digging a bit deeper the teslamate page is now working! but no data being displayed in grafana...
we're getting 'origin not allowed' - lots of proxy:error and proxy_wstunnel:error in the apache2 error log...

**Reply by ewrible** [reply]
So...  I have a similar issue.  When I add the key, I get...

eribble@NAS:/volume1/@docker$ sudo docker-compose down
ERROR: yaml.scanner.ScannerError: while scanning for the next token
found character '\t' that cannot start any token
  in "./docker-compose.yml", line 9, column 1

**Comment by mikeh2010** [comment, marked as answer]
Fixed - prob due to updating Ubuntu?? needed a ProxyPreserveHost in the teslamate.conf for grafana,,,
